test/jdk/javax/rmi/ssl/SSLSocketParametersTest.java

```diff
@@ -116,154 +130,104 @@ public ServerSocket createServerSocket(int port) throws IOException {
-                System.out.println(msg3);
-                System.exit(1);
-            } catch (Exception e) {
-                System.out.println(msg2 + " Exception: " + e.toString());
-                System.exit(0);
+
+            /* server side dummy_protocol */
+            case 5 -> {
+                try {
+                    new ServerFactory(null,
+                            new String[]{"dummy_protocol"}, false);
+                    throw new RuntimeException(
+                            "The expected exception for unsupported protocol was not thrown.");
+                } catch (IllegalArgumentException exc) {
+                    // expecting the exception for unsupported protocol,
+                    // anything else is an error
+                    if (!exc.getMessage().toLowerCase().contains("unsupported protocol")) {
+                        throw exc;
+                    }
+                }
             }
-            break;
-        case 6: /* client side dummy_ciphersuite */
-            System.out.println(msg1);
-            try {
+
+            /* client side dummy_ciphersuite */
+            case 6 -> {
                 System.setProperty("javax.rmi.ssl.client.enabledCipherSuites",
-                                   "dummy_ciphersuite");
-                HelloImpl server = new HelloImpl(
-                          0,
-                          new ClientFactory(),
-                          new ServerFactory());
-                Remote stub = server.runServer();
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg3);
-                System.exit(1);
-            } catch (Exception e) {
-                System.out.println(msg2 + " Exception: " + e.toString());
-                System.exit(0);
+                        "dummy_ciphersuite");
+                run(new ServerFactory(), true);
             }
-            break;
-        case 7: /* client side dummy_protocol */
-            System.out.println(msg1);
-            try {
+
+            /* client side dummy_protocol */
+            case 7 -> {
                 System.setProperty("javax.rmi.ssl.client.enabledProtocols",
-                                   "dummy_protocol");
-                HelloImpl server = new HelloImpl(
-                          0,
-                          new ClientFactory(),
-                          new ServerFactory());
-                Remote stub = server.runServer();
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg3);
-                System.exit(1);
-            } catch (Exception e) {
-                System.out.println(msg2 + " Exception: " + e.toString());
-                System.exit(0);
+                        "dummy_protocol");
+                run(new ServerFactory(), true);
             }
-            break;
-        default:
```

Comment:
The default should be retained to produce an error.

Reply:
I put the default case back